### PR TITLE
3.0,dev: Add cloud related tools documentation

### PR DESCRIPTION
**Pull request by aylei**
Signed-off-by: Aylei <rayingecho@gmail.com>

### What is changed, added or deleted? 
Add cloud related tools documentation

### Checklist 

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere 
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

@DanielZhangQD @xiaojingchen @lilin90 @tennix PTAL

### CODE REVIEW

**Comment by aylei** [comment]
@dcalvin PTAL

File: dev/reference/tools/tools-in-kubernetes.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    ```shell
+    kubectl port-forward -n <namespace> <tikv-pod-name> 20160:20160 &>/tmp/portforward-tikv.log
+    ```
+
+    打开连接后，即可通过本地的对应端口访问 PD 服务和 TiKV 节点：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    $ tikv-ctl --ca-path ca.pem --cert-path client.pem --key-path client-key.pem --host 127.0.0.1:20160 <subcommands>
```
Comment (DanielZhangQD) [inline dev/reference/tools/tools-in-kubernetes.md]: Where to get the pem files?

File: dev/reference/tools/tools-in-kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    ```
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    tikv-ctl --pd 127.0.0.1:2379 compact-cluster
+    ```
+
+* **本地模式**：本地模式需要访问 TiKV 的数据文件，并且需要停止正在运行的 TiKV 实例。需要先使用[诊断模式](/how-to/troubleshoot/tidb-in-kubernetes.md#诊断模式)关闭 TiKV 实例自动重启，关闭 TiKV 进程，再使用 `tkctl debug` 命令在目标 TiKV Pod 中启动一个包含 `tikv-ctl` 可执行文件的新容器来执行操作，步骤如下：
```
Comment (DanielZhangQD) [inline dev/reference/tools/tools-in-kubernetes.md]: In which case should users use local mode? Is it reasonable to use on production env?

File: dev/how-to/deploy/tidb-operator.md [hunk 3]
```diff
@@ -13,11 +13,11 @@ category: how-to
 TiDB Operator 部署前，请确认以下软件需求：
 
 * Kubernetes v1.10 或者更高版本
+* 如果要使用 zone-aware 持久化卷, 则需要 Kubernetes v1.12 或者更高版本
```
Comment (tennix) [inline dev/how-to/deploy/tidb-operator.md]: All our tests are based on 1.12, so I think it's better to require the minimal cluster version to 1.12.

File: dev/reference/tools/tools-in-kubernetes.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+执行上述命令后，就可以通过 `127.0.0.1:2379` 访问到 PD 服务，从而直接使用 `pd-ctl` 命令的默认参数执行操作，如：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+pd-ctl -d config show
+```
+
+假如你本地的 2379 被占据，则需要选择其它端口，此时需要为 `pd-ctl`  命令显式指定 PD 端口：
```
Comment (tennix) [inline dev/reference/tools/tools-in-kubernetes.md]: Add documents about how to use kubectl port-forward if port 2379 is occupied by another program.

File: dev/reference/tools/tools-in-kubernetes.md (hunk 1 above)
Comment (tennix) [inline dev/reference/tools/tools-in-kubernetes.md]: Currently, tidb-operator does not support TiKV data encryption and TiDB cluster inter-connection TLS. So the certificate is unnecessary.

File: dev/TOC.md [hunk 5]
```diff
@@ -291,6 +291,7 @@
   + [TiDB Binlog 简介](reference/tidb-binlog-overview.md)
   + 生态工具
     - [tkctl](reference/tools/tkctl.md)
+    - [Kubernetes 上的工具使用](reference/tools/tools-in-kubernetes.md)
```
Comment (xiaojingchen) [inline dev/TOC.md]: should we move `tkctl` to `tools-in-kubernetes.md`?

File: dev/reference/tools/tools-in-kubernetes.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+* `tidb-operator`：用于部署 TiDB Operator；
+* `tidb-cluster`：用于部署 TiDB 集群；
+* `tidb-backup`：用于 TiDB 集群备份恢复；
+
+这些 chart 都托管在 PingCAP 维护的 helm chart 仓库 `http://charts.pingcap.org/` 中，你可以通过下面的命令添加该仓库：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+helm repo add pingcap http://charts.pingcap.org/
```
Comment (tennix) [inline dev/reference/tools/tools-in-kubernetes.md]: Use `https`.

File: dev/reference/tools/tools-in-kubernetes.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+
+* 执行升级（升级可以是修改 `chart-version` 升级到新版本的 chart，也可以是修改 `values.yaml` 文件更新应用配置）：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    helm upgrade <release-name> <chart-name> --version=<chart-version> -f <values-file>
+    ```
+
+最后，假如要删除 helm 部署的应用，可以执行：
```
Comment (tennix) [inline dev/reference/tools/tools-in-kubernetes.md]: `helm ls` is useful before install/upgrade/delete.

File: dev/reference/tools/tools-in-kubernetes.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+helm del --purge <release-name>
+```
+
+更多 helm 的相关文档，请参考 [Helm 官方文档](https://helm.sh/docs/)。
+
+## 使用 Terraform
+
+[Terraform](https://www.terraform.io/) 是一个基础设施即代码管理工具。它允许用户使用声明式的风格描述自己的基础设施，并针对描述生成执行计划来创建或调整真实世界的计算资源。Kubernetes 上的 TiDB 使用 Terraform 来在公有云上创建和管理 TiDB 集群。
+
+你可以参考[https://github.com/helm/helm#install](https://www.terraform.io/downloads.html) 来安装 Terraform。
```
Comment (tennix) [inline dev/reference/tools/tools-in-kubernetes.md]: The URL is mixed with Helm URL.

File: dev/TOC.md (hunk 5 above)
Comment (aylei) [reply dev/TOC.md]: I suppose not, `tkctl` is the first-class **command line interface**, while this document describes the **related** tools

File: dev/reference/tools/tools-in-kubernetes.md (hunk 2 above)
Comment (aylei) [reply dev/reference/tools/tools-in-kubernetes.md]: It is covered in the [TiKV Control] document, and is not the concern of our document

**Comment by aylei** [comment]
@xiaojingchen @DanielZhangQD @tennix All comments are addressed, PTAL again

File: dev/how-to/deploy/tidb-operator.md [hunk 9]
```diff
@@ -34,7 +33,7 @@ TiDB Operator 运行在 Kubernetes 集群，你可以使用[这里](https://kube
 
 如果你要使用不同环境，必须在 Kubernetes 集群中安装 DNS 插件。可以根据[官方文档](https://kubernetes.io/docs/tasks/access-application-cluster/configure-dns-cluster/)搭建 DNS 插件。
 
-TiDB Operator 使用[持久化卷](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)持久化 TiDB 集群数据（包括数据库，监控和备份数据），所以 Kubernetes 集群必须提供至少一种持久化卷。为提高性能，建议使用本地 SSD 盘作为持久化卷。可以根据[这一步](#本地持久化卷)自动配置本地持久化卷。
+TiDB Operator 使用[持久化卷](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)持久化 TiDB 集群数据（包括数据库，监控和备份数据），所以 Kubernetes 集群必须提供至少一种持久化卷。为提高性能，建议使用本地 SSD 盘作为持久化卷。可以根据[这一步](#配置本地持久化卷)自动配置本地持久化卷。
```
Comment (dcalvin) [inline dev/how-to/deploy/tidb-operator.md]:
```suggestion
TiDB Operator 使用[持久化卷](https://kubernetes.io/docs/concepts/storage/persistent-volumes/)持久化存储 TiDB 集群数据（包括数据库，监控和备份数据），所以 Kubernetes 集群必须提供至少一种持久化卷。为提高性能，建议使用本地 SSD 盘作为持久化卷。可以根据[这一步](#配置本地持久化卷)自动配置本地持久化卷。
```

File: dev/reference/tools/tools-in-kubernetes.md [hunk 10]
```diff
@@ -0,0 +1,256 @@
+---
+title: Kubernetes 上的 TiDB 工具指南
+category: reference
+---
+
+# Kubernetes 上的 TiDB 工具指南
+
+Kubernetes 上的 TiDB 运维管理需要使用一些开源工具，同时，在 Kubernetes 上使用 TiDB 生态工具时，也有特殊的操作要求。本文档详细描述 Kubernetes 上的 TiDB 相关的工具及其使用方法。
```
Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
Kubernetes 上的 TiDB 运维管理需要使用一些开源工具。同时，在 Kubernetes 上使用 TiDB 生态工具时，也有特殊的操作要求。本文档详细描述 Kubernetes 上的 TiDB 相关的工具及其使用方法。
```

File: dev/reference/tools/tools-in-kubernetes.md [hunk 11]
```diff
@@ -0,0 +1,256 @@
...
+
+假如你本地的 2379 被占据，则需要选择其它端口：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl port-forward -n <namespace> svc/<cluster-name>-pd <local-port>:2379 &>/tmp/portforward-pd.log
+```
+
+此时，需要为 `pd-ctl`  命令显式指定 PD 端口：
```
Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
此时，需要为 `pd-ctl` 命令显式指定 PD 端口：
```

File: dev/reference/tools/tools-in-kubernetes.md [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+
+{{< copyable "shell-regular" >}}
+
+```shell
+pd-ctl -u 127.0.0.1:<local-port> -d config show
+```
+
+## 在 Kubernetes 上使用 TiKV Control
+
+[TiKV Control](reference/tools/tikv-control.md) 是 TiKV 的命令行工具，在使用 TiKV Control 操作 Kubernetes 上的 TiDB 集群时，针对 TiKV Control 的不同操作形式，有不同的操作步骤。
```
Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
[TiKV Control](reference/tools/tikv-control.md) 是 TiKV 的命令行工具。在使用 TiKV Control 操作 Kubernetes 上的 TiDB 集群时，针对 TiKV Control 的不同操作模式，有不同的操作步骤。
```

Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
[TiKV Control](reference/tools/tikv-control.md) 是 TiKV 的命令行工具。在使用 TiKV Control 操作 Kubernetes 上的 TiDB 集群时，针对 TiKV Control 的不同操作模式，有不同的操作步骤。
```

File: dev/reference/tools/tools-in-kubernetes.md [hunk 13]
```diff
@@ -0,0 +1,256 @@
...
+
+    如果 Kubernetes 集群没有启用 `RBAC`，那么可以直接使用下列命令安装 `tiller`：
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    helm init --upgrade
+    ```
+
+Kubernetes 应用在 helm 中被打包为 chart，PingCAP 针对 Kubernetes 上的 TiDB 部署运维提供了三个 helm chart：
```
Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
Kubernetes 应用在 helm 中被打包为 chart。PingCAP 针对 Kubernetes 上的 TiDB 部署运维提供了三个 Helm chart：
```

File: dev/reference/tools/tools-in-kubernetes.md [hunk 14]
```diff
@@ -0,0 +1,256 @@
...
+
+当新版本的 chart 发布后，你可以使用 `helm repo update` 命令更新本地对于仓库的缓存：
+
+{{< copyable "shell-regular" >}}
+
+```
+helm repo update
+```
+
+Helm 的常用操作有部署（`helm install`）、升级（`helm upgrade`)、销毁（`helm del`)、查询（`helm ls`）。Helm chart 往往都有很多可配置参数，通过命令行进行配置比较繁琐，因此推荐使用 YAML 文件的形式来编写这些配置项，基于 Helm 社区约定俗称的称呼方式，我们在文档中将用于配置 chart 的 YAML 文件称为 `values.yaml` 文件。
```
Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
Helm 的常用操作有部署（`helm install`）、升级（`helm upgrade`)、销毁（`helm del`)、查询（`helm ls`）。Helm chart 往往都有很多可配置参数，通过命令行进行配置比较繁琐，因此推荐使用 YAML 文件的形式来编写这些配置项，基于 Helm 社区约定俗称的命名方式，我们在文档中将用于配置 chart 的 YAML 文件称为 `values.yaml` 文件。
```

File: dev/reference/tools/tools-in-kubernetes.md [hunk 15]
```diff
@@ -0,0 +1,256 @@
...
+
+执行部署、升级、销毁等操作前，可以使用 `helm ls` 查看集群中已部署的应用：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+helm ls
+```
+
+在执行部署和升级操作时，必须指定使用的 chart 名字（`chart-name`）和部署后的应用名（`release-name`），还可以指定一个或多个 `values.yaml` 文件来配置 chart，此外，假如有特定的版本需求，则需要通过 `--version` 参数指定 `chart-version` (默认为最新的 GA 版本）。命令形式如下：
```
Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
在执行部署和升级操作时，必须指定使用的 chart 名字（`chart-name`）和部署后的应用名（`release-name`），还可以指定一个或多个 `values.yaml` 文件来配置 chart。此外，假如对 chart 有特定的版本需求，则需要通过 `--version` 参数指定 `chart-version` (默认为最新的 GA 版本）。命令形式如下：
```

File: dev/reference/tools/tools-in-kubernetes.md [hunk 16]
```diff
@@ -0,0 +1,256 @@
...
+
+```shell
+helm del --purge <release-name>
+```
+
+更多 helm 的相关文档，请参考 [Helm 官方文档](https://helm.sh/docs/)。
+
+## 使用 Terraform
+
+[Terraform](https://www.terraform.io/) 是一个基础设施即代码管理工具。它允许用户使用声明式的风格描述自己的基础设施，并针对描述生成执行计划来创建或调整真实世界的计算资源。Kubernetes 上的 TiDB 使用 Terraform 来在公有云上创建和管理 TiDB 集群。
```
Comment (dcalvin) [inline dev/reference/tools/tools-in-kubernetes.md]:
```suggestion
[Terraform](https://www.terraform.io/) 是一个基础设施即代码（Infrastructore as Code）管理工具。它允许用户使用声明式的风格描述自己的基础设施，并针对描述生成执行计划来创建或调整真实世界的计算资源。Kubernetes 上的 TiDB 使用 Terraform 来在公有云上创建和管理 TiDB 集群。
```

**Comment by aylei** [comment]
@dcalvin Addressed, PTAL

**Pull request merged**